src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -0,0 +1,216 @@
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+class Stats:
+    '''
+    Store statistics of the given path that'll be copied/deleted/purged (or
+    handled, in short) in an organized manner.
+    '''
+    def __init__(self, given_path, op, fsh, should_cancel):
+        # path for which this class instance will contain statistics
+        self.given_path = given_path
+
+        self.size_t = 0     # total size
+        self.rfiles_t = 0   # total regfiles
+        self.subdirs_t = 0  # total subdirectories
+        self.slinks_t = 0   # total symlinks
+
+        self.size_c = 0     # size copied
+        self.rfiles_c = 0   # num of regfiles copied
+        self.subdirs_c = 0  # num of dirs copied
+        self.slinks_c = 0   # num of symlinks copied
+
+        self.op = op
+        # creating self.oped so that we can pring log messages accordingly.
+        self.oped_str = self._get_operated_str() # oped = operated
+
+        self.get_statistics_for_path(fsh, should_cancel)
+
+    def get_statistics_for_path(self, fsh, should_cancel):
+        '''
+        Get statistics like of total number of files and subdirs of under
+        "given_path" and total size of data underneath "given_path".
+        '''
+        self.rfiles_t = int(fsh.getxattr(self.given_path, 'ceph.dir.rfiles'))
+        self.subdirs_t = int(fsh.getxattr(self.given_path,'ceph.dir.rsubdirs'))
+        self.size_t = int(fsh.getxattr(self.given_path, 'ceph.dir.rbytes'))
+
+        self.log_total_amount()
+
+        if should_cancel():
+            raise VolumeException(-errno.EINTR,
+                                  "user interrupted clone operation")
+
+    @property
+    def percent(self):
+        if self.size_t == 0:
+            return 100 if self.size_c == 0 else 0
+        return ((self.size_c/self.size_t) * 100)
+
+    @property
+    def progress_fraction(self):
+        '''
+        For progress MGR module (mgr/progress), value of attribute "progress"
+        should be in range of 0.0 to 1.0. 0.0 represents 0% progress and 1.0
+        represents 100% progress.
+        '''
+        return round(self.percent, 2)/100
+
+    def _get_operated_str(self):
+        '''
+        Return past tense form of the operation for which this instance is
+        being used.
+        '''
+        if self.op == 'copy':
+            return 'copied'
+        elif self.op == 'purge':
+            return 'purged'
+        else:
+            log.critical('Class Stats is being used for an operation it '
+                         'doesn\'t recognize')
+
+    def get_size_stat(self, human=True):
```

Comment:
I think you are referring to this one - https://github.com/ceph/ceph/pull/54620#discussion_r1428163841

Reply:
Do you want it exactly like `statx()`? Because we have `get_progress_report()` which returns all info one call - https://github.com/ceph/ceph/pull/54620/files#diff-c0d245e49ccb876cd07bfddafb303a1b366b8a2c70f1e16c9bfbd4f8c79cf11dR183